### ISSUE TITLE
ESP32 not receiving correct data from NRF24L01

### ISSUE THREAD

**Issue by rptmatrox** [nRF24/RF24Network#199]
First of all: Im sorry for my english. I´ll try to do my best. =)

Well, the story goes like this:
I have a homemade project for monitoring various parameters at home such as: temperature, humidity, atmospheric pressure, current, voltage, power, etc...
In the first version I used an Arduino MEGA 2560 (WEMOS WIFI) as a base and arduinos NANO as satellites with sensors. The data was sent to the BLYNK platform.
The routine is: every 15 seconds the base send a data request for each satellite and when they receive it, they send the sensor data back to the base. This way I would have a synchronized timestamp of all collected data.

**The first problem (SOLVED):** over time I had issues with the WIFI connection on the MEGA base and I opted to exchange it for an ESP32.
**The second problem (NOT SOLVED):** now I have a stable WIFI connection, but now I have other issue receiving data from the NRF24L01.

The code of NANOs is the same.
The code of BASE (MEGA vs ESP32) is almost the same except for the CE, CSN PIN configuration.

**To simplify the example and make the code cleaner I removed all the BLYNK integration, Wi-Fi, and some satellites and left only one set to you help me with the issue.**

### BASE CODE (ARDUINO MEGA and ESP32) *The Diference is the CE/ CSN Pin declaration comment or not:
(CODE IN TXT FORMAT BELOW)
[RF_NETWORK_MEGA_BASE_ESP32_TESTE.txt](https://github.com/nRF24/RF24Network/files/10319955/RF_NETWORK_MEGA_BASE_ESP32_TESTE.txt)

### ARDUINO NANO CODE (SAT2 reading temperature and Humidity):
(CODE IN TXT FORMAT BELOW)
[RF_NETWORK_NANO_SAT2.txt](https://github.com/nRF24/RF24Network/files/10319948/RF_NETWORK_NANO_SAT2.txt)

OK, the Serial results for each board:
ARDUINO NANO AS SAT2 - The sensors are reading correct values and the PAYLOAD to be sent is OK:
![2022-12-29_105350](https://user-images.githubusercontent.com/37125636/209964271-0f8a9f2b-16b6-4cbe-a2b5-189e26fa128b.png)

ARDUINO MEGA AS BASE - The payload value received is OK:
![2022-12-29_111010](https://user-images.githubusercontent.com/37125636/209965156-687156d1-f013-4331-9f04-0f9e11c85cab.png)

ESP32 AS BASE - The Value received is NOT OK:
![2022-12-29_115925](https://user-images.githubusercontent.com/37125636/209971206-b390b61a-8509-457a-ab58-e731059dc08a.png)

As you can see the payload received appear to be "trash".
But I think the radio is working fine, since the ESP32 shows the radio parameters, can send a request and the satelite node respond. Only the payload is strange. =/

### The hardware PIN configuration in ESP32 are:

GND..........GND
VCC 3.3....VCC 3.3
MOSI........D23 (2)
MISO........D19 (8)
SCK...........D18 (9)
CSN..........D5  (10)
CE.............D4  (13)
IRQ...........unplugged

**Comment by Avamander** [comment]
Are the integers in the struct the same size on both platforms and aligned identical in memory? (Using fixed-size data types and telling the compiler to *pack* the struct would be a really really good thing to so)

Are you calculating the size of the payloads correctly on both ends?

You can determine if the bug is in your code by taking the example sketches and by running those without modifications.

**Comment by rptmatrox** [comment]
@Avamander Thanks for your response.
After your comment I researched about it and found this topic that spoke exactly about this variable type issue:
https://forum.arduino.cc/t/data-transmision-issue-from-arduino-pro-mini-to-esp32-via-nrf24l01/918278/4

I tested in my code change the variables inside STRUCT to int32_t in both codes (TX and RX) and it worked. 
Like this:

```
struct payload_R_SAT2 {
  int32_t temperaturaI;
  int32_t umidadeI;
};
```

Thanks a LOT for the TIP.
Anyway this topic could be usefull for other users =)
